apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -41,7 +43,57 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         /// Credentials can be retrieved from a file or obtained interactively.
         /// See Google Cloud documentation for more information.
         /// </summary>
-        public ITokenAccess Credential { get; private set; }
+        public ITokenAccess GetCredential()
+        {
+            // Check the ctor override.
+            if (CredentialOverride != null)
+            {
+                return CredentialOverride;
+            }
+
+            //Calculate it from the CredentialFile argument in the connection string.
+            GoogleCredential credential = null;
+            string jsonFile = CredentialFile;
+            if (!string.IsNullOrEmpty(jsonFile))
+            {
+                if (string.Compare(Path.GetExtension(jsonFile), ".json", StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    throw new ArgumentException($"{nameof(CredentialFile)} should only be set to a JSON file.");
```

Comment:
This shouldn't be an ArgumentException, as it's a parameterless method. InvalidOperationException?

Reply:
sure